### ISSUE TITLE
Write for async write function finishes

### ISSUE THREAD

**Issue by efreibe** [felHR85/UsbSerial#238]
Hello,

I'm trying to open the usb port, send data and close it. But I realized that not all the data
is not transferred to destination.
I suspect that while writing I send a close command and my data is truncated.
How can I listen to know when all buffer was totally written to the port?

Thank you

**Comment by felHR85** [comment]
Hi @efreibe 
[I would use the synchronous api  in your case](https://github.com/felHR85/UsbSerial/wiki/5.-Synchronous-api)

**Comment by efreibe** [comment]
Yes, I'm doing that, I've created a thread and I'm making calls to syncWrite

I simply created the issue because it will be helpful for other people making async requests 
to get a notification when the write operation is done.

I'm using flutter and the easy way to send a success message back to the channel is waiting for a notification from write async method. But I managed to do it in other way.

You can close it, it's not a blocking issue.

**Comment by felHR85** [comment]
@efreibe  Like a JS promise with "then" added at the end to handle the result of the operation?
I guess it could be added because it is a nice idea.

**Comment by efreibe** [comment]
Good to hear that. It will give more control to the write process.
